File: grid-challenge.js
function gridChallenge(grid) {
    let rows = grid.length
    console.log(rows)
    
    for ( let i = 0; i < rows; i++) {
        grid[i] = grid[i].split('').sort().join('')
    }
    
    let columns = grid[0].length;
    
    for (let column = 0; column < columns; column++) {
        for (let row = 0; row < rows - 1; row++) {
            if (grid[row][column] > grid[row + 1][column]) {
                return "NO";
            }
        }
    }
    
    return "YES"

}

File: grid-challenge.py
def gridChallenge(grid):
    rows = len(grid)
    columns = len(grid[0])

    for i in range(rows):
        grid[i] = ''.join(sorted(grid[i]))

    for column in range(columns):
        for row in range(rows - 1):
            if grid[row][column] > grid[row + 1][column]:
                return "NO"

    return "YES"

